BookDAO.php

```diff
@@ -32,5 +32,48 @@ public function fetchAll(): array
         return $books;
     }
 
+    public function add(Book $book):int
+    {
+        $sql = 'INSERT INTO `books` (`title`, `author`, `genre`, `year`, '
+            . '`progressperc`, `rating10`, `coverlink`, `gr-link`) '
+            . 'VALUES (:title, :author, :genre, :year, :progressperc, :rating10, :coverlink, :grLink) ';
+
+        $values = [
+            'title' => $book->getTitle(),
+            'author' => $book->getAuthor(),
+            'genre' => $book->getGenre(),
+            'year' => $book->getYear(),
+            'progressperc' => $book->getProgressPercent(),
+            'rating10' => $book->getRating(),
+            'coverlink' => $book->getCoverLink(),
+            'grLink' => $book->getGrLink()
+        ];
+
+        $query = $this->db->prepare($sql);
+        $query->execute($values);
+
+        return $this->db->lastInsertId(); // Request the ID used and return
+    }
+
+    public function listOfAuthorsAndId(): array
+    {
+        $sql = 'SELECT `authors`.`id`, `name`'
+            . 'FROM `authors`;';
+        $query = $this->db->prepare($sql);
+        $query->execute();
+
+        return $query->fetchAll();
+    }
```

Comment:
Authors are not books!

While getting authors out is basically the same as getting books, they are two distinct entities. 

Rather than putting this method in this class, it would be more appropriate to create a new DAO class specifically for authors.